### PR TITLE
Moving icon to submenu when expanded

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

Outdated icon in vertical nav bar should be displayed in **Integration** when menu collapsed, then move to **Configuration** when menu expanded

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-4392

move-icon2

### CODE REVIEW

**Comment by josemigallas** [comment]
I think it would be better to handle this in JS, firstly by reverting the changes made in #1605 and then updating the predicates with `isOutOfDate && isExpanded`.

The problem with the CSS approach is that it's completely inflexible and make it not anymore extendable (say we needed to add more out-of-date icons to other sections).

WDYT @damianpm @didierofrivia ?

**Comment by thomasmaas** [comment]
I think that would be premature optimization. For me this perfect for we need now.

**Comment by thomasmaas** [comment]
To clarify my last statement, we should just restore the behavior that was in place: https://github.com/3scale/porta/pull/1605/files#diff-f82d17ff82ceac4cebb8068986ecb9e3L245

**Comment by thomasmaas** [comment]
This issue also needs a corresponding jira issue targeted at ER2 as QE might find regression in ER1.

**Comment by damianpm** [comment]
The problem with the js approach is that, previously, we were passing a prop to a react component, because it was our own component, not a PF standard component.
Now, using PF component, it doesn't expect this behavior, not a prop or a child component, to make it work we would need to start hacking the component. The css option is the less invasive.
@josemigallas

**Comment by josemigallas** [comment]
> Now, using PF component, it doesn't expect this behavior, not a prop or a child component, to make it work we would need to start hacking the component. The css option is the less invasive.

It doesn't have to be invasive, `NavItem`s can still be passed a class, then adding the icon via CSS.

**Comment by didierofrivia** [comment]
IMHO, since every PF component accept classes as params, the way to go would be:

1. Computing the warning state (js)
2. Passing it as class in the component class array (js)
3. Let css to manage the visual

File: app/assets/stylesheets/provider/_vertical_nav.scss [hunk 1]
```diff
@@ -7,12 +7,18 @@
...
   // scss-lint:disable SelectorFormat
-  .pf-c-nav__item.outdated-config::before {
-    content: '\f071';
-    font-family: FontAwesome;
-    font-weight: normal;
-    position: absolute;
-    right: line-height-times(3);
-    top: line-height-times(1 / 3);
+  .pf-c-nav__item {
+    &.outdated-config > a::before {
```
Comment (josemigallas) [inline app/assets/stylesheets/provider/_vertical_nav.scss]: If my suggestion below is rejected I think this one might improve readability:
```suggestion
    &.outdated-config:not(.pf-m-expanded) > a::before {
```

File: app/javascript/src/Navigation/components/VerticalNav.jsx [hunk 2]
```diff
@@ -39,9 +40,9 @@ const VerticalNav = ({ sections, activeSection, activeItem }: Props) => (
 const NavSection = ({title, isSectionActive, activeItem, items, outOfDateConfig}) => {
   return (
     <NavExpandable title={title} isActive={isSectionActive} isExpanded={isSectionActive} className={outOfDateConfig ? 'outdated-config' : ''}>
```
Comment (josemigallas) [inline app/javascript/src/Navigation/components/VerticalNav.jsx]: Only a suggestion
```suggestion
  const shouldShowIcon = outOfDateConfig && !isSectionActive
  return (
    <NavExpandable title={title} isActive={isSectionActive} isExpanded={isSectionActive} className={shouldShowIcon ? 'outdated-config' : ''}>
```

Comment (damianpm) [reply app/javascript/src/Navigation/components/VerticalNav.jsx]: It doesn't work, I already tried `isSectionActive`, but it only changes it's value when you click and navigate to any of the subitems, not when you expand/collapse from other pages

Comment (josemigallas) [reply app/javascript/src/Navigation/components/VerticalNav.jsx]: Right, it should be the state not the prop.

**Pull request merged**